diff --git a/tools/make_spec.lua b/tools/make_spec.lua
index 1ff19d9..865a845 100644
--- a/tools/make_spec.lua
+++ b/tools/make_spec.lua
@@ -1,8 +1,18 @@
 local lcmark = require('lcmark')
 local cmark = require('cmark')
+local lyaml = require('lyaml')
 
 local format = arg[1] or 'html'
 
+local jarefs = {}
+local jarefs_file = io.open("../refs.ja.yaml", "r")
+if jarefs_file then
+   jarefs = lyaml.load(jarefs_file:read("a"))
+   jarefs_file:close()
+else
+   io.stderr:write("WARNING: ja refs file not open.  Please rerun\n")
+end
+
 local trim = function(s)
   return s:gsub("^%s+",""):gsub("%s+$","")
 end
@@ -12,7 +22,15 @@ local warn = function(s)
 end
 
 local to_identifier = function(s)
-  return trim(s):lower():gsub('[^%w]+', ' '):gsub('[%s]+', '-')
+   s = trim(s)
+   id = jarefs[s]
+   if id then
+      return id
+   end
+   if s:find('[^\0-\127]') then
+      warn("not ASCII: " .. s)
+   end
+   return s:lower():gsub('[^%w]+', ' '):gsub('[%s]+', '-')
 end
 
 local render_number = function(tbl)
@@ -216,11 +235,11 @@ local create_anchors = function(doc, meta, to)
         local examplenum_div = make_html_block('div', {{'class', 'examplenum'}})
         local interact_link = make_html_inline('a', {{'class', 'dingus'},
                     {'title', 'open in interactive dingus'}})
-        cmark.node_append_child(interact_link, make_text("Try It"))
+        cmark.node_append_child(interact_link, make_text("やってみよう"))
         local examplenum_link = cmark.node_new(cmark.NODE_LINK)
         cmark.node_set_url(examplenum_link, '#example-' .. tostring(example))
         cmark.node_append_child(examplenum_link,
-                                make_text("Example " .. tostring(example)))
+                                make_text("例 " .. tostring(example)))
         cmark.node_append_child(examplenum_div, examplenum_link)
         if format == 'html' then
           cmark.node_append_child(examplenum_div, interact_link)
@@ -267,6 +291,10 @@ for lab, ident in pairs(refs) do
   end
 end
 refblock = refblock .. pluralrefblock
+if format == 'refblock' then
+  print(refblock)
+  os.exit(0)
+end
 -- append references and parse again
 local contents, meta, msg  = lcmark.convert(inp .. refblock, format,
                              { smart = true,
