diff --git a/tools/make_spec.lua b/tools/make_spec.lua
index 1ff19d9..865a845 100644
--- a/tools/make_spec.lua
+++ b/tools/make_spec.lua
@@ -267,6 +291,10 @@ for lab, ident in pairs(refs) do
   end
 end
 refblock = refblock .. pluralrefblock
+if format == 'refblock' then
+  print(refblock)
+  os.exit(0)
+end
 -- append references and parse again
 local contents, meta, msg  = lcmark.convert(inp .. refblock, format,
                              { smart = true,
